### DISCUSSION TITLE
No more Github Copilot for Russian students?

### DISCUSSION THREAD

**Discussion by sectapunterx** [community/community#47626] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have an active student pack on github education, but copilot is inactive. on the page https://github.com/github-copilot/signup I have the message: Cannot proceed with payment
There appears to be an issue with your account. We're looking into it and will send you an update within 48 hours. Thank you for your patience.
Is the student copilot no longer active?

**Comment by ivan-plotnikov** [comment]
It already happened a month ago. After a while it recovered on its own. I hope it will be the same this time. It seems that they turn off free users in case of heavy load.

**Comment by tuves** [comment]
Thank you for reporting this. The bug fix has now been deployed to production and affected users should be able to connect. 

Due to the recency of this incident, we are still investigating the contributing factors and will provide a more detailed update on cause and remediation in the February Availability Report, which we will publish the first Wednesday of March.

We sincerely apologize for any stress and/or frustration this has caused.

**Reply by vaskers5** [reply]
the same trouble here - I activated the students pack, but the github-copilot stopped working today

**Reply by ivan-plotnikov** [reply]
@vaskers5 

Is there still a problem?

**Reply by shredder67** [reply]
I think the bug is still there for large sample of users, primarly from RU, you might need to reinvestigate it. 

Although for me it's the issue related to 403 auth error, but I am using Student Pack too, so whatever, maybe it's somehow related.

**Reply by pkuderov** [reply]
Even though I got an email that my GitHub Copilot access has been renewed (through Students Pack as well), PyCharm Copilot plugin still is not able to log in.

(and thanks for continue supporting our education/research community despite all this hell)

**Reply by AkobArm** [reply]
At the moment my Copilot no working.The opening Pycharm and error Copilot this(At this time, Copilot is not available in your location)

**Reply by innokentiy2517** [reply]
Got email that my access has been renewed. But WS says, Copilot is not available in your country. Cool. Thanks for punishing innocent Russian students :D

**Reply by vaskers5** [reply]
> @vaskers5
> 
> Is there still a problem?

Now github copilot ts not available in your location

**Reply by ivan-plotnikov** [reply]
This morning the copilot started working. Works only through vpn. If you turn off the vpn, it works for a while, then it stops. Well, it's better than nothing.

**Reply by lartie** [reply]
Same issue

> Now github copilot ts not available in your location

**Reply by matwej29** [reply]
Still not working

**Reply by AliMilani** [reply]
same in **IRAN**

**Reply by WexxanBest** [reply]
> > @vaskers5
> > Is there still a problem?
> 
> Now github copilot ts not available in your location

Same issue. Also have an active student pack on github education. Having issue in VS Code extension saying:
```
GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"
```

I am from Russia.
It works fine with VPN.

![image](https://user-images.githubusercontent.com/57976769/219804243-b980207b-fa18-4574-8da8-e274cc3bd27f.png)

**Reply by riZZZhik** [reply]
Same issue in Russia \
With VPN everything works fine

```
GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"
```

**Reply by shredder67** [reply]
Can you guys share what VPNs use :) I have been using Cloudfare, but it doesnt help here

**Reply by lartie** [reply]
@tuves any updates?

**Reply by Akash1134** [reply]
Hi there,

Sorry for the difficulty you're experiencing. I'm afraid that due to current trade controls in some countries as listed [here](https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls) , GitHub account cannot make use of Copilot.

While GitHub obtained a license in 2021 to provide cloud services to Iranian users, Copilot is not included in the scope of this license.

I'm sorry if that wasn't what you were hoping to hear.

We want every developer to be able to collaborate on GitHub, and are undertaking efforts to increase Copilot availability around the world, so please know that GitHub will continue to advocate for the right of developers in trade-restricted regions to access tools for code collaboration and communication.

For more details, see [GitHub and Trade Controls](https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#can-you-clarify-availability-of-github-to-cuban-developers)

**Reply by 0dminnimda** [reply]
> Can you guys share what VPNs use :) I have been using Cloudfare, but it doesnt help here

Windscribe works nicely, 10 Gb is more than enough for everybody

**Reply by ra1nbow1** [reply]
@tuves what is with bug fixes?

**Comment by ra1nbow1** [comment]
That's awful. Copilot still unavailable. When the bug fixes will be deployed?

**Comment by 0dminnimda** [comment]
GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"

**Comment by tuves** [comment]
We became aware that Copilot was unavailable in several countries following our efforts to restore access to Ukraine, and on further investigation, we identified the cause as originating from a change we deployed to block Copilot in trade-restricted countries. To prevent this from happening again, we've updated our reporting dashboards to better account for error rates due to trade restrictions and are implementing more robust testing of the trade restriction logic.

**Reply by sectapunterx** [reply]
What does this mean? Will the copilot be unavailable in Russia?

**Reply by 0dminnimda** [reply]
> What does this mean? Will the copilot be unavailable in Russia?

Russia is on a list of trade-restricted countries, so yes

**Reply by ra1nbow1** [reply]
> > What does this mean? Will the copilot be unavailable in Russia?
> 
> Russia is on a list of trade-restricted countries, so yes

Can you send a screenshot with Russia in this list?

**Reply by sectapunterx** [reply]
> > > What does this mean? Will the copilot be unavailable in Russia?
> > 
> > 
> > Russia is on a list of trade-restricted countries, so yes
> 
> Can you send a screenshot with Russia in this list?

somewhere there was a response from a github representative with this phrase

**Reply by 0dminnimda** [reply]
@ra1nbow1 
https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#github-enterprise-server

**Reply by ra1nbow1** [reply]
GitHub Enterprise Server?

**Reply by 0dminnimda** [reply]
> GitHub Enterprise Server may not be sold to, exported, or re-exported to any country listed in Country Group E:1 in Supplement No. 1 to part 740 of the EAR or to the following regions of Ukraine: Crimea and the separatist areas of Donetsk and Luhansk. The countries listed currently include Cuba, Iran, North Korea, Syria, Russia, and Belarus, but this list is subject to change.

**Reply by 0dminnimda** [reply]
I am basing those replies on [this message](https://github.com/orgs/community/discussions/47626#discussioncomment-5051905)

**Reply by ra1nbow1** [reply]
I got, but there's nothing about GitHub copilot actually

**Reply by 0dminnimda** [reply]
AFAIK this is the list of trade-restricted countries

**Comment by sy-python** [comment]
Having the same problem.

**Comment by beliboba** [comment]
Same problem.

**Comment by Serge-sudo** [comment]
same problem)))

**Comment by ra1nbow1** [comment]
It works!!!

**Reply by sectapunterx** [reply]
without proxy/VPN?

**Reply by Serge-sudo** [reply]
It says, it is not available in your country)) 
with VPN works fine

**Reply by ra1nbow1** [reply]
> without proxy/VPN?

With VPN, sorry. I didn't notice my turned on VPN 😁

**Reply by Yoshisoul** [reply]
@ra1nbow1 

which vpn do you use?

**Reply by ra1nbow1** [reply]
Surfshark

**Comment by sectapunterx** [comment, marked as answer]
I created a new topic some time ago, in which I said that Copilot in Russia does not work without VPN/proxy, because it is not available in this country
For JetBrains IDE you need to use http proxy, socks does not help

**Reply by lookandhate** [reply]
Can you suggest any worth proxy to use copilot with?

**Reply by sectapunterx** [reply]
I bought http proxies in the telegram channel, some of the cheapest for 16 rubles/month. however, I do not use them anywhere for data safety except copilot

**Reply by ra1nbow1** [reply]
> I bought http proxies in the telegram channel, some of the cheapest for 16 rubles/month. however, I do not use them anywhere for data safety except copilot

Can you send a link please?

**Reply by beliboba** [reply]
@sectapunterx Link please. Pycharm just wont work every time i try with proxies that i bought

**Comment by nirname** [comment]
This is still not working, unfortunately

**Comment by mazzdriver** [comment]
It looks like real racism, not financial or any other circumstances when people are restricted in something just because of where they were born or how they were born. It seems that recently we were moving away from this, Nuremberg is still in memory.
Why can't I use the service from my regular account, without any use of bypassing locks?
